certora/specs/SafeWebAuthnSignerSingleton.spec

```diff
@@ -1,51 +1,92 @@
+using WebAuthnHarness as WebAuthnHarness;
+
+methods {
+    function P256.verifySignatureAllowMalleability(P256.Verifiers a, bytes32 b, uint256 c, uint256 d, uint256 e, uint256 f) internal returns (bool) => 
+        verifySignatureAllowMalleabilityGhost(a, b, c, d, e, f);
+
+    function WebAuthn.encodeSigningMessage(bytes32 challenge, bytes calldata authenticatorData, string calldata clientDataFields) internal returns (bytes memory) =>
+        GETencodeSigningMessageCVL(challenge, authenticatorData, clientDataFields);
+
+    function WebAuthnHarness.checkInjective(bytes32 challenge, bytes32 authenticatorData, bytes32 clientDataFields, bytes32 result) internal returns (bool) =>
+        checkInjectiveSummary(challenge, authenticatorData, clientDataFields, result);
+}
+
+function GETencodeSigningMessageCVL(bytes32 challenge, bytes authenticatorData, string clientDataFields) returns bytes
+{
+    env e;
+    return WebAuthnHarness.GETencodeSigningMessageSummary(e, challenge, authenticatorData, clientDataFields);
+}
+
+ghost checkInjectiveSummary(bytes32, bytes32, bytes32, bytes32) returns bool {
+    axiom forall bytes32 x1. forall bytes32 y1. forall bytes32 z1. forall bytes32 x2. forall bytes32 y2. forall bytes32 z2. forall bytes32 result.
+    checkInjectiveSummary(x1, y1, z1, result) && checkInjectiveSummary(x2, y2, z2, result) => x1 == x2;
```

Comment:
Why we don't require that also:
`y1 == y2 && z1 == z2` ?

Or is there some assumption that we need to mention?

Reply:
As Safe asked, given input in encodeSigningMessage(), one should produce only one valid output. i.e., one cannot reuse a signature valid for one challenge to be valid for another challenge